kornia/geometry/epipolar/numeric.py

```diff
@@ -24,3 +24,613 @@ def cross_product_matrix(x: torch.Tensor) -> torch.Tensor:
+                b: a first order polynomial for variables :math:`(x,y,z,1)`.
+
+    Returns:
+        degree 2 poly with the order :math:`(x^2, x*y, x*z, x, y^2, y*z, y, z^2, z, 1)`.
+    """
+
+    return torch.stack(
+        [
+            a[:, 0] * b[:, 0],
+            a[:, 0] * b[:, 1] + a[:, 1] * b[:, 0],
+            a[:, 0] * b[:, 2] + a[:, 2] * b[:, 0],
+            a[:, 0] * b[:, 3] + a[:, 3] * b[:, 0],
+            a[:, 1] * b[:, 1],
+            a[:, 1] * b[:, 2] + a[:, 2] * b[:, 1],
+            a[:, 1] * b[:, 3] + a[:, 3] * b[:, 1],
+            a[:, 2] * b[:, 2],
+            a[:, 2] * b[:, 3] + a[:, 3] * b[:, 2],
+            a[:, 3] * b[:, 3],
+        ],
+        dim=-1,
+    )
+
+
+# Reference
+# Nistér, David. An efficient solution to the five-point relative pose problem. 2004.
+# https://github.com/danini/graph-cut-ransac/blob/aae1f40c2e10e31fd2191bac601c53a189673f60/src/pygcransac/
+# include/estimators/solver_essential_matrix_five_point_nister.h#L156
+
+
+def o2(a: torch.Tensor, b: torch.Tensor) -> torch.Tensor:
+    r"""Multiply two polynomials a and b of degrees two and one.
+
+    Args:
+        a: a second degree poly for variables :math:`(x^2, x*y, x*z, x, y^2, y*z, y, z^2, z, 1)`.
+        b: a first degree poly for variables :math:`(x y z 1)`.
+
+    Returns:
+        a third degree poly for variables,
+        :math:`(x^3, y^3, x^2*y, x*y^2, x^2*z, x^2, y^2*z, y^2,
+        x*y*z, x*y, x*z^2, x*z, x, y*z^2, y*z, y, z^3, z^2, z, 1)`.
+    """
+
+    return torch.stack(
+        [
+            a[:, 0] * b[:, 0],
+            a[:, 4] * b[:, 1],
+            a[:, 0] * b[:, 1] + a[:, 1] * b[:, 0],
+            a[:, 1] * b[:, 1] + a[:, 4] * b[:, 0],
+            a[:, 0] * b[:, 2] + a[:, 2] * b[:, 0],
+            a[:, 0] * b[:, 3] + a[:, 3] * b[:, 0],
+            a[:, 4] * b[:, 2] + a[:, 5] * b[:, 1],
+            a[:, 4] * b[:, 3] + a[:, 6] * b[:, 1],
+            a[:, 1] * b[:, 2] + a[:, 2] * b[:, 1] + a[:, 5] * b[:, 0],
+            a[:, 1] * b[:, 3] + a[:, 3] * b[:, 1] + a[:, 6] * b[:, 0],
+            a[:, 2] * b[:, 2] + a[:, 7] * b[:, 0],
+            a[:, 2] * b[:, 3] + a[:, 3] * b[:, 2] + a[:, 8] * b[:, 0],
+            a[:, 3] * b[:, 3] + a[:, 9] * b[:, 0],
+            a[:, 5] * b[:, 2] + a[:, 7] * b[:, 1],
+            a[:, 5] * b[:, 3] + a[:, 6] * b[:, 2] + a[:, 8] * b[:, 1],
+            a[:, 6] * b[:, 3] + a[:, 9] * b[:, 1],
+            a[:, 7] * b[:, 2],
+            a[:, 7] * b[:, 3] + a[:, 8] * b[:, 2],
+            a[:, 8] * b[:, 3] + a[:, 9] * b[:, 2],
+            a[:, 9] * b[:, 3],
+        ],
+        dim=-1,
+    )
+
+
+# Compute degree 10 poly representing determinant (equation 14 in the paper)
+# https://github.com/danini/graph-cut-ransac/blob/aae1f40c2e10e31fd2191bac601c53a189673f60/src/pygcransac/
+# include/estimators/solver_essential_matrix_five_point_nister.h#L368C5-L368C82
+def det_to_poly(A: torch.Tensor) -> torch.Tensor:
```

Comment:
```suggestion
def determinant_to_polynomial(A: Tensor) -> Tensor:
```
more explicit names and `from kornia.core import Tensor`